docs/libs/ml/optimization.md

```diff
@@ -184,12 +192,15 @@ mini-batch SGD will be performed.
         </td>
       </tr>
       <tr>
-        <td><strong>Prediction Function</strong></td>
+        <td><strong>ConvergenceThreshold</strong></td>
         <td>
           <p>
-            Class that provides the prediction function, used to calculate $\hat{y}$ based on the 
-            weights and the example features, and the prediction gradient.
-            (Default value: <strong>LinearPrediction</strong>)
+            When set, this value acts as a convergence stopping criterion for the algorithm.
+            If the relative difference in the objective function (loss) between two consecutive
+            iterations
+            is less than this value we stop the iterations and return the last weights.
+            This allows us to avoid unnecessary iterations after the algorithm has converged.
+            (Default value: <strong>None</strong>)
```

Comment:
Maybe we could specify the convergence criterion also as a formula, then it's clear which value is used to calculate the relative change. Maybe we can make a list with all supported convergence schemes (at the moment only the relative change one).